### DISCUSSION TITLE
Hi i need to download the libraries exeption and socket???

### DISCUSSION THREAD

**Discussion by vianneycol** [senaranya/HL7#112] [General]
Hi,  where i can to download the libraries  exeption and socket??? i need to connected to bio chemestry analyzer. Can you help me???

**Comment by vianneycol** [comment]
![Uploading error.jpg…]()

**Comment by senaranya** [comment]
You need to use [composer](https://getcomposer.org/) to download and install the package. There are tutorials available online/YouTube on how to use composer to install a php package.

Let me know if that still doesn't work for you

**Reply by vianneycol** [reply]
Thanks, i will tell you if when finish all the process. Again thanks.

**Comment by vianneycol** [comment]
Hi, greetings to all. I already installed the libraries, I am using VisualStudio Code as an editor. Then I need to receive the results from a blood chemistry equipment, the equipment is using the IP 192.168.1.150 and port 1001, therefore I assume the following process:

1) load the libraries:
require "./vendor/autoload.php";

use Aranyasen\HL7\Connection;
use Aranyasen\HL7\Message;
use Aranyasen\HL7\Response;
use Aranyasen\Exceptions\Exception;
use Aranyasen\Exceptions\HL7ConnectionException;


2) I open the connection:
$Ip_Equipment="192.168.1.150";
$Poort_Equipmen=1001;

$connection = new Connection($Ip_Equipment, $Poort_Equipmen);
$req_recived = new Message();


3) sending a response ACK message:

$msg_confir_response = new Message("MSH|^~\\&|1|\rABC|1||^AAAA1^^^BB|", null, true);
$ackResponse = new ACK($msg_confir_response);
$ack_final = (new Connection($Ip_Equipment, $Poort_Equipmen))->send($msg_confir_response);
4) finally I convert the received message to text:

$req_recived->toString(); 

Please confirm to me if this process would be correct. thanks again for your help.

Vianney


[Correo Aranyasen 10_12_2023.docx](https://github.com/senaranya/HL7/files/13629450/Correo.Aranyasen.10_12_2023.docx)

**Reply by senaranya** [reply]
This library can't be used as a listener unfortunately. So point 2 in your example won't work as you're expecting. You need to use a separate listener/server that'll run on the background, accepts any incoming HL7 message and passes on to your code. Your code can then process the message and send an ACK like you've shown in 3

For a discussion on HL7 listener, please check #114
